src/views/AddItem.jsx

```diff
@@ -1,25 +1,44 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase.js';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data }) {
 	const [itemName, setItemName] = useState('');
 	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
-	const [itemAdded, setItemAdded] = useState(false);
 	const [error, setError] = useState(false);
+	const [userAlertMessage, setUserAlertMessage] = useState('');
 
-	// TODO: dynamically pass in the list token so item is added to correct list
 	const handleSubmit = async (e) => {
 		e.preventDefault();
+		const normalizedItemName = itemName.replace(/[\s\W]|_+/g, '').toLowerCase();
```

Comment:
Prob unnecessary but it may be helpful to leave a comment behind in the code explaining what the regex is doing here.

Reply:
there are options here, comment works, you can also, depending on the complexity of the matching required in the file, create an expression, name it something meaningful, and test it. I often name my expressions like phoneRegex, dateRegex, or addressRegex.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/RegExp
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

Reply:
Great suggestions! This could definitely be a constant, e.g. `const NORMALIZED_ITEM_NAME = <regex>`.

Reply:
Thanks for the feedback and suggestions here! The regex has been moved to a const, which hopefully will clarify what the regex is doing.